geomstats/geometry/base.py

```diff
@@ -320,32 +320,33 @@ class LevelSet(Manifold, abc.ABC):
     ----------
     dim : int
         Dimension of the embedded manifold.
-    embedding_space : VectorSpace
-        Embedding space.
     default_coords_type : str, {'intrinsic', 'extrinsic', etc}
         Coordinate type.
         Optional, default: 'intrinsic'.
     """
 
-    def __init__(
-        self,
-        dim,
-        embedding_space,
-        submersion,
-        value,
-        tangent_submersion,
-        default_coords_type="intrinsic",
-        **kwargs
-    ):
-        kwargs.setdefault("shape", embedding_space.shape)
+    def __init__(self, dim, default_coords_type="intrinsic", **kwargs):
+        kwargs.setdefault("shape", self.embedding_space.shape)
         super().__init__(dim=dim, default_coords_type=default_coords_type, **kwargs)
-        self.embedding_space = embedding_space
-        self.embedding_metric = embedding_space.metric
-        self.submersion = submersion
-        if isinstance(value, float):
-            value = gs.array(value)
-        self.value = value
-        self.tangent_submersion = tangent_submersion
+
+    @abc.abstractmethod
+    def _define_embedding_space(self):
+        pass
+
+    @abc.abstractmethod
+    def submersion(self, point):
+        """Submersion that defines the manifold."""
```

Comment:
Maybe add that the value should be zero, potentialy with a equation: submersion(x) = 0 defines the manifold